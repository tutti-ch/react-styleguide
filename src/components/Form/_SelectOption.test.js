/* global describe, test, expect, jest */
import React from "react"
import SelectOption from "./_SelectOption"
import { mount } from "enzyme"

describe("(Component) SelectOption", () => {
  test("[select] should trigger the onClick prop", () => {
    const onClick = jest.fn()
    const event = {}

    const comp = mount(
      <SelectOption
        onClick={onClick}
        highlighted={true}
        selected={true}
        icon="ico ico-tutti-cube"
        text="my option"/>
    )
    const inst = comp.instance()

    inst.select(event)
    expect(onClick).toHaveBeenCalledWith(inst.props, event)
    expect(comp).toMatchSnapshot()
  })
})