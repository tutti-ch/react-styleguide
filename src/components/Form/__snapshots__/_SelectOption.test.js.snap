// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SelectOption [select] should trigger the onClick prop 1`] = `
<Option
  highlighted={true}
  icon={<Icon />}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "highlighted": true,
            "icon": <Icon />,
            "onClick": [MockFunction] {
              "calls": [Circular],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "selected": true,
            "text": "my option",
          },
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  selected={true}
  text="my option"
>
  <div
    className=""
    onClick={[Function]}
  >
    <Icon>
      my-icon
    </Icon>
    <span>
      my option
    </span>
  </div>
</Option>
`;
